sklearn/preprocessing/_discretization.py

```diff
@@ -267,7 +278,28 @@ def fit(self, X, y=None):
                 bin_edges[jj] = np.linspace(col_min, col_max, n_bins[jj] + 1)
 
             elif self.strategy == "quantile":
-                quantiles = np.linspace(0, 100, n_bins[jj] + 1)
+                # if sample_weight (weight) parameter is defined, then the size
+                # of each bin should be proportional to each weight value (in w)
```

Comment:
You can remove the comment and the weight is applied for each sample not the bin

Reply:
I corrected this in https://github.com/scikit-learn/scikit-learn/commit/85fe2b74ee5fe571b92f113f9dd8182339a4c4aa